

表达量的纠正

|      | A2     | D5     | AD1_At | AD1_Dt | AD2_At | AD2_Dt |
| ---- | ------ | ------ | ------ | ------ | ------ | ------ |
| FPKM | 654903 | 702336 | 299964 | 278169 | 289458 | 296373 |

|        | A2_vs_AD1 | A2_vs_AD2 | D5_vs_AD1 | D5_vs_AD2 |
| ------ | --------- | --------- | --------- | --------- |
| 纠正数 | 2.183     | 2.263     | 2.525     | 2.370     |

提取基因表达量

```
#A2_vs_AD1
sh ~/script/get_different_exp.sh  -H ~/The_Crop_Journal/homologous/A2_vs_AD1.gene -s ~/The_Crop_Journal/RNA_seq/stringtie/A2/A2 -S ~/The_Crop_Journal/RNA_seq/stringtie/AD1/AD1 -i con_BA_PAV.bed -g ~/data/A2/A2_re_only_gene.bed -o . -p con_BA_PAV -t A2_vs_AD1

```



## 保守和不保守边界范围内同源基因的表达量的差异

### 全基因组水平

```bash
sh ~/script/get_different_exp.sh -H ~/The_Crop_Journal/homologous/A2_vs_AD1.gene -s ~/The_Crop_Journal/RNA_seq/stringtie/A2/A2 -S ~/The_Crop_Journal/RNA_seq/stringtie/AD1/AD1 -i A2_vs_AD1_con_boundary.txt -g ~/data/A2/A2_re_only_gene.bed -o . -p A2_vs_AD1_con -t A2_vs_AD1	#统计保守的基因表达量差异

sh ~/script/get_different_exp.sh -H ~/The_Crop_Journal/homologous/A2_vs_AD1.gene -s ~/The_Crop_Journal/RNA_seq/stringtie/A2/A2 -S ~/The_Crop_Journal/RNA_seq/stringtie/AD1/AD1 -i A2_vs_AD1_no_con_boundary.txt -g ~/data/A2/A2_re_only_gene.bed -o . -p A2_vs_AD1_con -t A2_vs_AD1	#统计不保守的基因表达量差异
```



|           | 保守的边界存在表达差异的同源基因数 | 保守的边界所有同源基因数 | 不保守的边界存在表达差异的同源基因数 | 不保守的边界所有同源基因数 | 全基因组所有存在表达差异的同源基因/总同源基因 |
| --------- | ---------------------------------- | ------------------------ | ------------------------------------ | -------------------------- | --------------------------------------------- |
| A2_vs_AD1 | 746(22.14%)                        | 3369                     | 1007(23.54%)                         | 4277                       | 6360/27467(23.15%)                            |
| A2_vs_AD2 | 768(28.4%)                         | 2704                     | 1462(29.19%)                         | 5009                       | 8263/27697(29.83%)                            |
| D5_vs_AD1 | 881(21.68%)                        | 4064                     | 503(20.41%)                          | 2465                       | 6231/28570(21.81%)                            |
| D5_vs_AD2 | 1017(29.9%)                        | 3405                     | 900(28.89%)                          | 3115                       | 7236/28582(25.32%)                            |

显著性测验

```R
R
table=matrix(c(3782,4796,827,1120),nrow=2,ncol=2)
chisq.test(table)
```

+ 全基因组水平上没有差异

## 结构变异对基因表达量的影响

```bash
#得到结构变异侧翼区间
awk '$3-$2>10000{print $0}' ~/The_Crop_Journal/SVs/A2_vs_AD1/A2_vs_AD1_A2_SV.bed |awk '{print $1"\t"$2-10000"\t"$2}{print $1"\t"$3"\t"$3+10000}' |sort -k1,1 -k2,3n >A2_vs_AD1_10K_SV.bed
awk '$3-$2>10000{print $0}' ~/The_Crop_Journal/SVs/A2_vs_AD2/A2_vs_AD2_all_SV_A2.bed |awk '{print $1"\t"$2-10000"\t"$2}{print $1"\t"$3"\t"$3+10000}' |sort -k1,1 -k2,3n >A2_vs_AD2_10K_SV.bed
awk '$3-$2>10000{print $0}' ~/The_Crop_Journal/SVs/D5_vs_AD1/D5_vs_AD1_all_D5_SV.bed |awk '{print $1"\t"$2-10000"\t"$2}{print $1"\t"$3"\t"$3+10000}' |sort -k1,1 -k2,3n >D5_vs_AD1_10K_SV.bed
awk '$3-$2>10000{print $0}' ~/The_Crop_Journal/SVs/D5_vs_AD2/D5_vs_AD2_all_D5_SV.bed |awk '{print $1"\t"$2-10000"\t"$2}{print $1"\t"$3"\t"$3+10000}' |sort -k1,1 -k2,3n >D5_vs_AD2_10K_SV.bed
#进行表达量统计
sh ~/script/get_different_exp.sh -H ~/The_Crop_Journal/homologous/A2_vs_AD1.gene -s ~/The_Crop_Journal/RNA_seq/stringtie/A2/A2 -S ~/The_Crop_Journal/RNA_seq/stringtie/AD1/AD1 -i A2_vs_AD1_10K_SV.bed -g ~/data/A2/A2_re_only_gene.bed -o . -p A2_vs_AD1 -t A2_vs_AD1
sh ~/script/get_different_exp.sh -H ~/The_Crop_Journal/homologous/A2_vs_AD2.gene -s ~/The_Crop_Journal/RNA_seq/stringtie/A2/A2 -S ~/The_Crop_Journal/RNA_seq/stringtie/AD2/AD2 -i A2_vs_AD2_10K_SV.bed -g ~/data/A2/A2_re_only_gene.bed -o . -p A2_vs_AD2 -t A2_vs_AD2
sh ~/script/get_different_exp.sh -H ~/The_Crop_Journal/homologous/D5_vs_AD1.gene -s ~/The_Crop_Journal/RNA_seq/stringtie/D5/D5 -S ~/The_Crop_Journal/RNA_seq/stringtie/AD1/AD1 -i D5_vs_AD1_10K_SV.bed -g ~/data/D5/D5_re_only_gene.bed -o . -p D5_vs_AD1 -t D5_vs_AD1
sh ~/script/get_different_exp.sh -H ~/The_Crop_Journal/homologous/D5_vs_AD2.gene -s ~/The_Crop_Journal/RNA_seq/stringtie/D5/D5 -S ~/The_Crop_Journal/RNA_seq/stringtie/AD2/AD2 -i D5_vs_AD2_10K_SV.bed -g ~/data/D5/D5_re_only_gene.bed -o . -p D5_vs_AD2 -t D5_vs_AD2

awk '$5>0{print $1"\t"$2"\t"$3"\t"$4"\t"log($5)/log(2)}$5==0{print $0}' SV_flanking_gene_diff_exp.txt |sort -k5n |awk '{print $5}' >vis.txt
```

差异检测

```
module load R/3.3.1-foss-2016b
R
a<-read.table("A2_vs_AD1_gene_diff_exp.txt")
wilcox.test(Paired=T,a[,2],a[,4])
```

## 三维结构对基因表达量的影响



example

|               |          |          |          |      |
| ------------- | -------- | -------- | -------- | ---- |
| D5_vs_AD1_D5  | 21160000 | 21600000 | 22240000 | 10   |
| D5_vs_AD1_AD1 | 21680000 | 22080000 | 22760000 | 10   |
|               |          |          |          |      |

```
/public/home/yxlong/miniconda3/envs/jupyter/lib/python3.7/site-packages/tadlib/visualize

from tadlib.visualize.heatmaps import *
vis = Triangle('/public/home/yxlong/subject/HiC/D5/TAD/rep1/D5.40000.cool::40000', '10', 21160000, 22240000)
vis.matrix_plot()
vis.plot_TAD('/public/home/yxlong/The_Crop_Journal/HiC/TAD/domain/D5_only_TAD.txt', linewidth=1.5)
vis.outfig('D5.jpg')

from tadlib.visualize.heatmaps import *
vis = Triangle('/public/home/yxlong/subject/HiC/AD1/TAD/AD1_Dt/rep2/AD1_Dt.40000.cool::40000', '10', 21680000, 22760000)
vis.matrix_plot()
vis.plot_TAD('/public/home/yxlong/The_Crop_Journal/HiC/TAD/domain/AD1_Dt_only_TAD.txt', linewidth=1.5)
vis.outfig('AD1.jpg')
```



### 2、计算表达量

|                         | A2_vs_AD1 |      |      |      |
| ----------------------- | --------- | ---- | ---- | ---- |
| All_con_BA_PAV_gene     | 83        |      |      |      |
| Diff_con_BA_PAV_gene    | 33        |      |      |      |
| All_no_con_BA_PAV_gene  | 140       |      |      |      |
| Diff_no_con_BA_PAV_gene | 55        |      |      |      |



### example

保守边界：结构变异只有一个，但基因表达量差异很大

```bash
cut -f8 1 | xargs -I {} grep {} ~/The_Crop_Journal/homologous/A2_vs_AD1.gene | cut -f1 | xargs -I {} grep {} 1 > 3	#寻找同源基因
cut -f8 3 | xargs -I {} grep {} ../A2_vs_AD1_con_gene_diff_exp.txt | awk '{print $5}' > 3_1	#计算表达量
paste 3 3_1 |sort|uniq > 4
awk '$9>=1||$9<=-1{print $1"\t"$2"\t"$3}' 4|uniq -c|sort -k1nr|less	#统计每个边界存在的差异表达基因的数目
```

```
A2_vs_AD1	Chr08	5840000	7080000	
			Chr08	7080000	7680000
```

不保守边界：

### 可视化

```
#提取基因注释
grep -f geneid ~/data/A2/A2_gene.gff3|awk '$3=="gene"'|awk 'BEGIN{print "molecule" "\t" "gene" "\t" "start" "\t" "end" "\t" "strand" "\t" "direction"}{a=substr($9,4,14)}{if($7=="+"){print "A2" "\t" a "\t" $4 "\t" $5 "\t" "forward" "\t" "1"}else{print "A2" "\t" a "\t" $4 "\t" $5 "\t" "reverse" "\t" "-1"}}' >gene.txt
```

